### DISCUSSION TITLE
AttributeError: 'module' object has no attribute 'WIZNET5K'

### DISCUSSION THREAD

**Discussion by skylin008** [micropython/micropython#12870] [STM32 / Pyboard]
I used the micropython V1.22, I rebuild the firmware support the network and w5500 as follow:  
make MICROPY_PY_LWIP=1 MICROPY_PY_NETWORK_WINZNET5K=5500 BOARD=PYBV11  -j , buit when run the follow code shows error: 

def __init__(self,  maxtry = 100):
        _spi = SPI(1, baudrate = 21_000_000)    # SPI1 highest speed for 42M
        _cs  = Pin('E_SS', Pin.OUT)
        _rst = Pin('E_RST', Pin.OUT)
        _rst(0)
        sleep_ms(10)
        _rst(1)
        self._maxtry = maxtry
        self._isRuning = False
        self.nic = network.WIZNET5K(_spi, _cs, _rst)

**Comment by jimmo** [comment]
> `make MICROPY_PY_LWIP=1 MICROPY_PY_NETWORK_WINZNET5K=5500 BOARD=PYBV11 -j `

WIZNET is misspelt.

**Reply by skylin008** [reply]
Oh , Thanks @jimmo

**Reply by skylin008** [reply]
I had modified the mistake of misspelt. But Can't found the WIZNET5K from network. How to solve this issue.

**Reply by jimmo** [reply]
Did you do a clean build?

**Reply by jimmo** [reply]
Definitely works here:

```
$ make -j BOARD=PYBV11 MICROPY_PY_NETWORK_WIZNET5K=5500 deploy
...
$ mpremote
Connected to MicroPython at /dev/ttyACM0
Use Ctrl-] or Ctrl-x to exit this shell
MicroPython v1.22.0-preview.118.g841422817e on 2023-11-03; PYBv1.1 with STM32F405RG
Type "help()" for more information.
>>> import network
>>> network.WIZNET5K
<class 'WIZNET5K'>
>>> 
```

**Reply by jimmo** [reply]
I noticed you're also enabling LWIP (instead of using the wiznet in mac raw mode). Also works:

```
$ make -j BOARD=PYBV11 MICROPY_PY_LWIP=1 MICROPY_PY_NETWORK_WIZNET5K=5500 deploy
...
$ mpremote
Connected to MicroPython at /dev/ttyACM0
Use Ctrl-] or Ctrl-x to exit this shell
MicroPython v1.22.0-preview.118.g841422817e on 2023-11-03; PYBv1.1 with STM32F405RG
Type "help()" for more information.
>>> 
>>> import network
>>> network.WIZNET5K
<class 'WIZNET5K'>
>>> import socket
>>> socket
<module 'lwip'>
```

(The last line confirms it's LWIP mode)

**Reply by skylin008** [reply]
Thanks @jimmo , I  will be try again.
